templates/fine-tune-llm_v2/cookbooks/optimize_cost/README.md

```diff
@@ -0,0 +1,232 @@
+| --- | --- | --- | --- | --- |
+| 512 | 64 | 101.47 | 0.47 | 0.74 |
+| 1024 | 32 | 103.7 | 0.48 | 0.72 |
+| 2048 | 8 | 99.75 | 0.47 | 0.71 |
+| 4096 | 4 | 102.58 | 0.48 | 0.69 |
+
+#### Llama-2-70B
+
+| Context Length | Bsize per device | Token Throughput -- TT (MT/hr) | MFU | Effective Cost ($/MT) |
+| --- | --- | --- | --- | --- |
+| 512 | 16 | 11.23 | 0.53 | 3.05 |
+| 1024 | 8 | 9.65 | 0.45 | 4.77 |
+| 2048 | 4 | 8.58 | 0.40 | 4.25 |
+| 4096 | 2 | 13.40 | 0.63 | 3.65 |
+
+#### Mixtral-8x7B
+
+| Context Length | Bsize per device | Token Throughput -- TT (MT/hr) | MFU | Effective Cost ($/MT) |
+| --- | --- | --- | --- | --- |
+| 512 | 64 | 59.73 | 0.56 | 0.41 |
+| 1024 | 32 | 57.20 | 0.53 | 0.40 |
+| 2048 | 16 | 56.85 | 0.53 | 0.40 |
+| 4096 | 8 | 55.84 | 0.52 | 0.40 |
+
+
+
+## Configurations
+
+In this section we cover a set of configurations that allow more customization in throughput optimization. Visit [docs](https://docs.anyscale.com/reference/finetuning-config-api) for more info. 
+
+
+## Instance type
+
+You can modify the cluster configuration by changing `worker_resources` and `trainer_resources`.
+
+The following is how you specify different GPU types in the YAML:
+
+| GPU type  | Resource Specification in YAML |
+|-----------|--------------------------------|
+| A10       | accelerator_type:A10G: 0.001  |
+| L4        | accelerator_type:L4: 0.001    |
+| A100-40G* | accelerator_type:A100-40G: 0.001 |
+| A100-80G* | accelerator_type:A100-80G: 0.001 |
+| H100*     | accelerator_type:H100: 0.001  |
+
+\* subject to availability.
+
+
+For instance if I want to use the example config `./training_configs/lora/llama-3-70b.yaml` and run a 70B model on A100-80G I have to change the following parts:
+
+```
+num_devices: 8
+trainer_resources: {}
+worker_resources:
+  accelerator_type:A100-80G: 0.001
+```
+
+All of our default configs are setup for A10G machines due to their more accesibility.
+
+#### Difference between `worker_resources` and `trainer_resources`
+
+In LoRA training you can often ignore `trainer_resources` and just provide the GPU type required for doing training under `worker_resources` similar to the example above.
+
+However, for full-parameter training on hetergoneous clusters (e.g. training 70B on A10G GPUs) it is a bit more convoluted. This is generally not recommended, but when you do not have access to X100 machines this is the only way you can still train super large models (fp16 and full-parameter).
+
+Ray Train allows users to specify a different set of resources for rank-0 vs. other ranks. Rank-0 is responsible for checkpoining and normally needs more CPU RAM than the other workers at the time of checkpointing, because of the implementation details around weight aggregation. In a heterogenous multi-node setting where you have both small nodes and large nodes with the same GPUs this can cause a problem, because it becomes important where rank-0 is stored and the cluster is not symetrically used all the time. 
```

Comment:
```suggestion
Ray Train allows users to specify a different set of resources for rank-0 vs. other ranks. Rank-0 is responsible for checkpointing and normally needs more CPU RAM than the other workers at the time of checkpointing, because of the implementation details around weight aggregation. In a heterogenous multi-node setting where you have both small nodes and large nodes with the same GPUs this can cause a problem, because it becomes important where rank-0 is stored and the cluster is not symmetrically used all the time. 
```